src/Microsoft.Tye.Extensions/Zipkin/ZipkinExtension.cs

```diff
@@ -14,48 +14,73 @@ internal sealed class ZipkinExtension : Extension
     {
         public override Task ProcessAsync(ExtensionContext context, ExtensionConfiguration config)
         {
-            if (context.Operation == ExtensionContext.OperationKind.LocalRun)
+            if (context.Application.Services.Any(s => s.Name == "zipkin"))
             {
-                if (context.Application.Services.Any(s => s.Name == "zipkin"))
-                {
-                    context.Output.WriteDebugLine("zipkin service already configured. Skipping...");
-                }
-                else
+                context.Output.WriteDebugLine("zipkin service already configured. Skipping...");
+            }
+            else
+            {
+                context.Output.WriteDebugLine("Injecting zipkin service...");
+                var service = new ContainerServiceBuilder("zipkin", "openzipkin/zipkin")
```

Comment:
Injecting zipkin as a service is useful for deployment, because then it can show up as a binding, and tye already knows how to prompt for it.

Basically we can model diagnostics sinks as *external* services and a lot of things *just work*.